flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/TwoPhaseCommitSinkFunction.java

```diff
@@ -362,4 +483,39 @@ public void setContext(Optional<CONTEXT> context) {
 			this.context = context;
 		}
 	}
+
+	/**
+	 * Adds metadata (currently only the start time of the transaction) to the transaction object.
+	 */
+	@VisibleForTesting
+	static class TransactionHolder<TXN> {
```

Comment:
`TransactionWrapper`?

Reply:
I will leave it as is. It's an internal class and it is clear what it does.